sources/packages/web/src/components/students/applicationTracker/ApplicationProgressBar.vue

```diff
@@ -0,0 +1,152 @@
+    >
+      <template #tick-label="{ tick, index }">
+        <span
+          v-if="index === trackerApplicationStatus"
+          class="label-bold default-color"
+          >{{ tick.label }}
+          <v-icon
+            v-if="hasDeclinedCard"
+            icon="fa:fas fa-exclamation-circle"
+            :size="20"
+            color="danger"
+            class="pl-4"
+        /></span>
+        <span class="label-value default-color" v-else>{{ tick.label }} </span>
+      </template>
+    </v-slider>
+
+    <draft
+      @editApplication="$emit('editApplication')"
+      v-if="applicationStatus === ApplicationStatus.draft"
+    />
+    <!-- The below components are checked with applicationStatusTracker[trackerApplicationStatus], so that in future if we need to see the previous, it can be easily attained just by removing readonly param from the v-slider or by adding a simple logic. -->
+    <submitted
+      v-if="
+        trackerApplicationStatus !== undefined &&
+        applicationStatusTracker[trackerApplicationStatus] ===
+          ApplicationStatus.submitted
+      "
+    />
+    <in-progress
+      v-if="
+        trackerApplicationStatus !== undefined &&
+        applicationStatusTracker[trackerApplicationStatus] ===
+          ApplicationStatus.inProgress
+      "
+      :applicationId="applicationId"
+      @declinedEvent="declinedEvent"
+    />
+  </template>
+  <cancelled v-else :applicationId="applicationId" />
+</template>
+<script lang="ts">
+import { ApplicationStatus } from "@/types";
+import { PropType, ref, defineComponent, computed } from "vue";
+import Draft from "@/components/students/applicationTracker/Draft.vue";
+import Submitted from "@/components/students/applicationTracker/Submitted.vue";
+import InProgress from "@/components/students/applicationTracker/InProgress.vue";
+import Cancelled from "@/components/students/applicationTracker/Cancelled.vue";
+
+export default defineComponent({
+  emits: ["editApplication"],
+  components: {
+    Draft,
+    Submitted,
+    InProgress,
+    Cancelled,
+  },
+  props: {
+    applicationId: {
+      type: Number,
+      required: true,
+    },
+    applicationStatus: {
+      type: Object as PropType<ApplicationStatus>,
+      required: true,
+    },
+  },
+  setup(props) {
+    const hasDeclinedCard = ref(false);
+    const applicationStatusTracker = ref<Record<number, ApplicationStatus>>({
+      0: ApplicationStatus.submitted,
+      1: ApplicationStatus.inProgress,
+      2: ApplicationStatus.assessment,
+      3: ApplicationStatus.enrollment,
+      4: ApplicationStatus.completed,
+    });
+    // trackFillColor will vary when more status are added.
+    const trackFillColor = ref("warning");
+
+    const applicationStatusOrder = (status: ApplicationStatus) => {
+      const [key] =
+        Object.entries(applicationStatusTracker.value).find(
+          ([, value]) => value === status,
+        ) ?? [];
+
+      if (key !== undefined) {
```

Comment:
You can also do this if(key).

Reply:
there is a key = 0, so, if(0) is false

Reply:
Sure, got it. You are correct. Thanks for the clarification